### PR TITLE
[nanoMIPS] Add eh_frame start and end symbols

### PR DESCRIPTION
**Pull request by AndrijaSyrmia**
eh_frame start and end symbols need to be defined if there is an
  eh_frame section. These symbols are needed by libunwind when used
  without libgcc.

### CODE REVIEW

File: gold/testsuite/nanomips_eh_frame_start_end_syms.s [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+_start:
+    addiu $a1, $a2, 1
+
+    .end _start
+    .cfi_endproc
+    .size _start, .-_start
+
+    .extern __eh_frame_start
+    .extern __eh_frame_end
+    # If not referenced, eh frame star/end symbols are not added
```
Comment (farazs-github) [inline gold/testsuite/nanomips_eh_frame_start_end_syms.s]: s/star/start

File: gold/nanomips.cc [hunk 2]
```diff
@@ -6643,6 +6654,69 @@ Target_nanomips<size, big_endian>::do_finalize_sections(
...
+    {
+      this->define_start_end_symbols("__eh_frame_start",
+            "__eh_frame_end",
+            symtab,
+            eh_frame_section);
+      Output_section* eh_frame_hdr_section =
+            layout->find_output_section(".eh_frame_hdr");
+
+      if (eh_frame_hdr_section != NULL)
+  this->define_start_end_symbols("__eh_frame_hdr_start",
```
Comment (farazs-github) [inline gold/nanomips.cc]: indentation

File: gold/testsuite/nanomips_eh_frame_start_end_syms.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+# nanomips_eh_frame_start_end_syms.sh -- test generation of start/end eh_frame
+# section symbols.
+
+# Copyright (C) 2018 Free Software Foundation, Inc.
```
Comment (farazs-github) [inline gold/testsuite/nanomips_eh_frame_start_end_syms.sh]: Fix year

**Review by farazs-github — APPROVED** [review]
Approved, with minor typos.

**Review by milica-lazarevic — APPROVED** [review]
LGTM with a small comment that it may be nice to reference libunwind at least in the commit message since we only need those symbols when relying on libunwind but not libgcc

**Comment by AndrijaSyrmia** [comment]
> LGTM with a small comment that it may be nice to reference libunwind at least in the commit message since we only need those symbols when relying on libunwind but not libgcc

Can you take a look if the new message is better, or should I be more verbose. If that is the case I'll have to see in detail why we need the symbols.

**Comment by milica-lazarevic** [comment]
> Can you take a look if the new message is better, or should I be more verbose. 

It's OK now.

**Pull request merged**